src/lib/context/shared/context.service.ts

```diff
@@ -58,6 +74,29 @@ export class ContextService {
     this.context$.next(context);
   }
 
+  private updateContextWithQueryParams(context: DetailedContext) {
+    return this.route.queryParams
```

Comment:
Wouldn't if be better if this subscription was in the service constructor and then it could simply update the current context?